From 722268d7bbb7eb78f35ad99137b30fc7e23503a6 Mon Sep 17 00:00:00 2001
From: "Panwar, Ashish" <ashish.panwar@intel.com>
Date: Fri, 4 Mar 2016 11:14:28 +0530
Subject: [PATCH 3954/4301] ASoC: Intel: Skylake: Wake up any potential reader
 after copying log data

commit 7511fa5c47fd5fe582e923ea2f28692b0f32e078 from
git://git@github.com:01org/linux-intel-4.9.git

Change-Id: I0f9587d0c7f6bdd83b5b0cd62afc51f767cffb92
Signed-off-by: Panwar, Ashish <ashish.panwar@intel.com>
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/8332
Reviewed-by: Babu, Ramesh <ramesh.babu@intel.com>
Tested-by: Babu, Ramesh <ramesh.babu@intel.com>
---
 sound/soc/intel/skylake/skl-fwlog.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/sound/soc/intel/skylake/skl-fwlog.c b/sound/soc/intel/skylake/skl-fwlog.c
index 281f8b0..f772b00 100644
--- a/sound/soc/intel/skylake/skl-fwlog.c
+++ b/sound/soc/intel/skylake/skl-fwlog.c
@@ -121,6 +121,7 @@ void skl_dsp_write_log(struct sst_dsp *sst, void __iomem *src, int core,
 		data++;
 	}
 	buff->total_avail += count;
+	wake_up(&buff->stream->runtime->sleep);
 }
 
 int skl_dsp_copy_log_user(struct sst_dsp *sst, int core,
-- 
1.7.5.4

